maven-scm-api/src/main/java/org/apache/maven/scm/command/changelog/AbstractChangeLogCommand.java

```diff
@@ -76,13 +84,21 @@ public ScmResult executeCommand( ScmProviderRepository repository, ScmFileSet fi
 
         ScmBranch branch = (ScmBranch) parameters.getScmVersion( CommandParameter.BRANCH, null );
 
+        ScmVersion version = parameters.getScmVersion( CommandParameter.SCM_VERSION, null );
+
         ScmVersion startVersion = parameters.getScmVersion( CommandParameter.START_SCM_VERSION, null );
 
         ScmVersion endVersion = parameters.getScmVersion( CommandParameter.END_SCM_VERSION, null );
 
         String datePattern = parameters.getString( CommandParameter.CHANGELOG_DATE_PATTERN, null );
 
-        if ( startVersion != null || endVersion != null )
+        boolean fromVersionToStartOfRepository = startVersion == null && endVersion == null && version != null;
```

Comment:
The variable name is a bit confusing because we have `startVersion` and the name contains `start` too.